clash-prelude/src/Clash/Class/HasDomain/HasSingleDomain.hs

```diff
@@ -173,6 +173,7 @@ type instance TryDomain t Double                = 'NotFound
 type instance TryDomain t (Maybe a)             = TryDomain t a
 type instance TryDomain t (Either a b)          = Merge t a b
 
+{-# DEPRECATED HasSingleDomain "Experimental feature multiple hidden has been removed. This type family will therefore be removed in Clash 1.12." #-}
```

Comment:
Please also slap `DEPRECATED` on `TryDomain`, so people who declare type instances for that family will also get a deprecation notice.

I wonder if we shouldn't just slap it on all types; I don't know which types people would actually _use_ in their code; I am certain that this goes for `TryDomain` but why not slap it on every type we will remove?

Reply:
Done.

Reply:
I've slapped the deprecation on the whole module, which I should have done in the first place.